service/src/main/java/org/broadinstitute/listener/relay/http/TargetHttpResponse.java

```diff
@@ -42,14 +44,27 @@ public OutputStream getCallerResponseOutputStream() {
   }
 
   public static TargetHttpResponse createTargetHttpResponseFromException(
-      int statusCode, Exception ex, RelayedHttpListenerContext context) {
+      int statusCode, Throwable ex, RelayedHttpListenerContext context) {
 
     String statusDescription = "";
     if (ex != null && ex.getMessage() != null) {
       statusDescription = ex.getMessage();
     }
 
-    return new TargetHttpResponse(null, null, statusCode, statusDescription, context);
+    String body =
+        String.format(
+            "{ \"message\":\"The listener failed to process the request.\", \"tracking_id\":\"%s\"}",
+            context.getTrackingContext().getTrackingId());
```

Comment:
shall we add the exception message to the response as well?

Reply:
I prefer not to return internal errors back to clients due to security concerns, also it may be meaningless to an end-user using the browser.  To help troubleshoot the issue a tracking id is provided. We can use it as a correlation id in the logs and see the details of what happened.

Reply:
that sounds good....hopefully we'll be able to setup logging in a way that it'll be easy to access listener log